source/crud/tests/unified/estimatedDocumentCount.yml

```diff
@@ -265,3 +265,65 @@ tests:
                 count: *collection0Name
               commandName: count
               databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with comment on 4.9.0 or greater"
+    runOnRequirements:
+      - minServerVersion: "4.9.0"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          comment: &objectComment { key: "value"}
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                aggregate: *collection0Name
+                pipeline: *pipeline
+                comment: *objectComment
+              commandName: aggregate
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with comment on less than 4.9.0"
+    runOnRequirements:
+      # https://jira.mongodb.org/browse/SERVER-63315
+      # Server supports count with comment of any type for comment starting from 4.4.14.
+      - minServerVersion: "4.4.14"
+        maxServerVersion: "4.8.99"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          comment: *objectComment
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                count: *collection0Name
+                comment: *objectComment
+              commandName: count
+              databaseName: *database0Name
+
+  - description: "estimatedDocumentCount with comment on less than 4.4.14 - string comment"
+    runOnRequirements:
+      - minServerVersion: "4.4.0"
+        maxServerVersion: "4.4.13"
+    operations:
+      - name: estimatedDocumentCount
+        object: *collection0
+        arguments:
+          comment: &stringComment "comment"
+        expectResult: 3
+    expectEvents:
+      - client: *client0
+        events:
+          - commandStartedEvent:
+              command:
+                count: *collection0Name
+                comment: *stringComment
+              commandName: count
+              databaseName: *database0Name
```

Comment:
I know we have _some_ tests for server-side errors using the `comment` option (e.g. `insertOne`). Can we add one more test here?

> estimatedDocumentCount with string comment - pre 4.4

This can be modeled after the "insertOne with comment - pre 4.4" test.

Reply:
Done

Reply:
I noted that an error test for string comments was not implemented, per the comment you left in the document comment test expecting a server-side error.